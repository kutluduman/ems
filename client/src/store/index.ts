import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import auth from "./slices/auth";
import employee from "./slices/employee";

export const store = configureStore({
  reducer: {
    auth,
    employee,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;