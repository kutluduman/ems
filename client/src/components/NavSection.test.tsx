import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import { store } from "../store";
import NavSection from "./NavSection";

describe("NavSection Component", () => {
  test("renders the NavSection Component", () => {
    const tree = create(
      <Provider store={store}>
        <Router>
          <NavSection open={true} navConfig={[]} />
        </Router>
      </Provider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});