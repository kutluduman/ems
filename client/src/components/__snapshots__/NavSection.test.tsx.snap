// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavSection Component renders the NavSection Component 1`] = `
<div
  className="MuiBox-root css-0"
  open={true}
>
  <ul
    className="MuiList-root css-1mk9mw3-MuiList-root"
  />
</div>
`;
